sql/api/src/main/scala/org/apache/spark/sql/types/DoubleType.scala

```diff
@@ -40,34 +38,4 @@ class DoubleType private() extends FractionalType {
  * @since 1.3.0
  */
 @Stable
-case object DoubleType extends DoubleType {
-
-  // Traits below copied from Scala 2.12; not present in 2.13
```

Comment:
In the previous PR that added these traits, were just these traits added for Scala 2.12/2.13? I am not sure if the object DoubleAsIfIntegral should be removed too